src/lib.rs

```diff
@@ -78,47 +93,13 @@ impl UPClientZenoh {
     /// ```
     /// #[tokio::main]
     /// # async fn main() {
-    /// use up_client_zenoh::UPClientZenoh;
-    /// use up_rust::{Number, UAuthority, UEntity, UUri};
-    /// use zenoh::config::Config;
-    /// let uauthority = UAuthority {
-    ///     name: Some("MyAuthName".to_string()),
-    ///     number: Some(Number::Id(vec![1, 2, 3, 4])),
-    ///     ..Default::default()
-    /// };
-    /// let uentity = UEntity {
-    ///     name: "default.entity".to_string(),
-    ///     id: Some(u32::from(rand::random::<u16>())),
-    ///     version_major: Some(1),
-    ///     version_minor: None,
-    ///     ..Default::default()
-    /// };
-    /// let upclient = UPClientZenoh::new(Config::default(), uauthority, uentity)
+    /// use up_transport_zenoh::{Config, UPClientZenoh};
+    /// let upclient = UPClientZenoh::new(Config::default(), String::from("MyAuthName"))
     ///     .await
     ///     .unwrap();
     /// # }
     /// ```
-    pub async fn new(
-        config: Config,
-        uauthority: UAuthority,
-        uentity: UEntity,
-    ) -> Result<UPClientZenoh, UStatus> {
-        // Validate the UUri
-        uauthority.validate_micro_form().map_err(|e| {
-            let msg = format!("UAuthority is invalid: {e:?}");
-            log::error!("{msg}");
-            UStatus::fail_with_code(UCode::INVALID_ARGUMENT, msg)
-        })?;
-        uentity.validate_micro_form().map_err(|e| {
-            let msg = format!("UEntity is invalid: {e:?}");
-            log::error!("{msg}");
-            UStatus::fail_with_code(UCode::INVALID_ARGUMENT, msg)
-        })?;
-        let source_uuri = UUri {
-            authority: Some(uauthority).into(),
-            entity: Some(uentity).into(),
-            ..Default::default()
-        };
+    pub async fn new(config: Config, authority_name: String) -> Result<UPClientZenoh, UStatus> {
```

Comment:
Hey @evshary -- for the uStreamer use-case we'll need a function to build a UPClientZenoh object from a Zenoh Runtime in order to fit within the topology with up-linux-streamer as a Zenoh plugin.

I'm okay if you want to merge this one and address that in a follow up. :+1:

Reply:
OK. Don't make the PR to lengthy. I'll create an issue to track.